src/zarr/abc/store.py

```diff
@@ -19,7 +19,22 @@
 
 __all__ = ["ByteGetter", "ByteSetter", "Store", "set_or_delete"]
 
-ByteRangeRequest: TypeAlias = tuple[int | None, int | None]
+
+class OffsetRange(TypedDict):
+    """Request all bytes starting from a given byte offset"""
+
+    offset: int
+    """The byte offset for the offset range request."""
+
+
+class SuffixRange(TypedDict):
+    """Request up to the last `n` bytes"""
+
+    suffix: int
+    """The number of bytes from the suffix to request."""
+
+
+ByteRangeRequest: TypeAlias = None | tuple[int, int] | OffsetRange | SuffixRange
```

Comment:
Is there a likely future case where `ByteRangeRequest = None` would not be supported, such that we should exclude `None` from the TypeAlias options?

Reply:
What does `None` mean? Can the implication of each type be documented in a docstring on this type?

Reply:
I think `None` would mean that the entire object is requested.

Reply:
IMO it's slightly clearer to have `None` outside this type alias. I like how it was before https://github.com/zarr-developers/zarr-python/blob/12f601258d7af950a853a9e7fbbdc32feae73901/src/zarr/abc/store.py#L144
where `ByteRangeRequest` defines a _valid_ byte range request, and then the type is used as `ByteRangeRequest | None`.

Reply:
Good call thanks